src/components/landing/pages/register/register.tsx

```diff
@@ -0,0 +1,154 @@
+import React, { FormEvent, useEffect, useState } from 'react'
+import { useSelector } from 'react-redux'
+import { Redirect } from 'react-router'
+import { doInternalRegister } from '../../../../api/auth'
+import { ApplicationState } from '../../../../redux'
+import { getAndSetUser } from '../../../../utils/apiUtils'
+import { Row, Col, Card, Form, Button, Alert } from 'react-bootstrap'
+import { Trans, useTranslation } from 'react-i18next'
+import { TranslatedExternalLink } from '../../../common/links/translated-external-link'
+import { ShowIf } from '../../../common/show-if/show-if'
+
+export enum RegisterError {
+  NONE,
+  USERNAME_EXISTING,
+  OTHER
+}
+
+export const Register: React.FC = () => {
+  const { t } = useTranslation()
+  const config = useSelector((state: ApplicationState) => state.config)
+  const user = useSelector((state: ApplicationState) => state.user)
+
+  const [username, setUsername] = useState('')
+  const [password, setPassword] = useState('')
+  const [passwordAgain, setPasswordAgain] = useState('')
+  const [error, setError] = useState(RegisterError.NONE)
+  const [ready, setReady] = useState(false)
+
+  const doAsyncRegister = async (): Promise<void> => {
+    await doInternalRegister(username, password)
+    await getAndSetUser()
+  }
+
+  const doRegisterSubmit = (event: FormEvent): void => {
+    doAsyncRegister().catch((err: Error) => {
+      if (err.message === RegisterError[RegisterError.USERNAME_EXISTING]) {
```

Comment:
Ohh okay.. now i get it. hm. You can put strings as values into enums. What if you use the i18n keys of the errors for the enum values and put them into the error state?

Reply:
Done in b4303af4